src/core/audio/AudioJack.cpp

```diff
@@ -286,11 +298,17 @@ void AudioJack::renamePort(AudioPort* port)
 }
 
 
+int AudioJack::setBufferSizeCallback(jack_nframes_t nframes, void* udata)
```

Comment:
Do you know if this is allowed to do allocations? I just ask because some callbacks must happen in realtime-context and thus are not allowed to do slow things like `new[]`.

My guess is that realtime-safety does not matter when a buffer size changes, but I could not tell from the docs.

Reply:
According to Perplexity AI it is called in a non-realtime thread. However, I wasn't able to find that information in the pointers it gave either. :thinking: So it should be taken with a big grain of salt.

On the other hand, it would be strange if that information was communicated on the realtime audio thread because it is very likely that clients need to allocate memory as well if that event happens.

<details><summary>Perplexitiy blurb</summary>
Yes, the callback that informs about changed buffer sizes in the JACK audio API is called outside of the audio thread. This callback, known as the buffer size callback, is executed in a separate non-real-time thread[1][2].

The buffer size callback is registered using the `jack_set_buffer_size_callback` function. It is called whenever the size of the buffer that will be passed to the process callback is about to change[1]. This callback is designed to allow clients that depend on knowing the buffer size to prepare for the change before it occurs.

It's important to note that:

1. The buffer size callback is not called from within the real-time audio processing thread.
2. The normal process cycle is suspended during the execution of this callback, causing a gap in the audio flow[2].
3. Because it's not in the real-time thread, the callback can perform operations that are not real-time safe, such as allocating memory or touching previously unreferenced memory[2].

This design allows for more flexibility in handling buffer size changes without interfering with the real-time audio processing thread, which has strict timing requirements.

Citations:
[1] https://jackaudio.org/api/group__ClientCallbacks.html
[2] https://jackclient-python.readthedocs.io/en/0.3.0/
[3] https://github.com/spatialaudio/jackclient-python/issues/81
[4] https://docs.rs/jack/latest/jack/struct.Client.html
[5] https://stackoverflow.com/questions/78429446/which-thread-handles-a-realtime-callback-in-jack-audio/78882670
[6] https://linuxmusicians.com/viewtopic.php?t=22620
[7] https://devforum.play.date/t/c-api-sound-callbacks-happen-on-another-thread/4414
[8] https://www.rncbc.org/drupal/node/2162
</details>